### PR TITLE
add metadata fields for nydusify filesystem check

### PR DESCRIPTION
**Pull request by wraymo**
#435 Part 1
Signed-off-by: wraymo <raywangv@gmail.com>

### CODE REVIEW

**Comment by yqleng1987** [comment]
@wraymo , a new test job has been submitted. Please wait in patience.

**Comment by yqleng1987** [comment]
@wraymo , The CI test is completed, please check result:
<table>
    <tr>
        <th colspan="2">Test Case</th><th>Test Result</th>
    </tr>
    <tr>
        <td colspan="2">merge-target-branch</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/6063>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">build-docker-image</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/6063>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-nydus</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/6063>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-ctr-remote</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/6063>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-nydus-snapshotter</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/6063>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">start-nydus-snapshotter-config-containerd</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/6063>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">run-container-with-nydus-image</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/6063>:white_check_mark:SUCCESS</a></td>
    </tr>
</table>

Congratulations, your test job passed!

**Comment by yqleng1987** [comment]
@wraymo , your pull request has been updated. A new test job will be submitted. Please wait in patience.

**Comment by yqleng1987** [comment]
@wraymo , your test job has passed, and no need to test again.

**Comment by yqleng1987** [comment]
@wraymo , your pull request has been updated. A new test job will be submitted. Please wait in patience.

**Comment by yqleng1987** [comment]
@wraymo , the test job has been submitted. Please wait in patience.

**Comment by yqleng1987** [comment]
@wraymo , The CI test is completed, please check result:
<table>
    <tr>
        <th colspan="2">Test Case</th><th>Test Result</th>
    </tr>
    <tr>
        <td colspan="2">merge-target-branch</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/6796>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">build-docker-image</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/6796>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-nydus</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/6796>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-ctr-remote</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/6796>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">compile-nydus-snapshotter</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/6796>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">start-nydus-snapshotter-config-containerd</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/6796>:white_check_mark:SUCCESS</a></td>
    </tr>
    <tr>
        <td colspan="2">run-container-with-nydus-image</td><td><a href=https://tone.openanolis.cn/ws/nrh4nnio/test_result/6796>:white_check_mark:SUCCESS</a></td>
    </tr>
</table>

Congratulations, your test job passed!

File: contrib/nydusify/pkg/checker/rule/filesystem.go [hunk 1]
```diff
@@ -114,7 +121,22 @@ func (rule *FilesystemRule) walk(rootfs string) (map[string]Node, error) {
 			// Ignore directory size check
 			size = info.Size()
 		}
+
 		mode := info.Mode()
+		var symlink string
+		if mode&os.ModeSymlink == os.ModeSymlink {
+			if symlink, err = os.Readlink(path); err != nil {
+				return err
```
Comment (imeoer) [inline contrib/nydusify/pkg/checker/rule/filesystem.go:129]: Better to wrap error with `errors.Wrapf(err, "read link %s", path)` :)

**Review by imeoer — COMMENTED** [review]
Thanks for the contribution! Others LGTM.

**Comment by yqleng1987** [comment]
@wraymo , your test job has passed, and no need to test again.

**Review by imeoer — APPROVED** [review]

**Pull request merged**